src/VersionParser.php

```diff
@@ -21,16 +24,22 @@ class VersionParser
     private $versions;
 
     /**
+     * @var Composer\VersionParser
+     */
+    private $parser;
+
+    /**
      * @var string
      */
-    private $modifier = '[._-]?(?:(stable|beta|b|RC|alpha|a|patch|pl|p)(?:[.-]?(\d+))?)?([.-]?dev)?';
+    const GIT_DATA_MATCH = '/.*(-\d+-g[[:alnum:]]{7})$/';
 
     /**
      * @param array $versions
      */
     public function __construct(array $versions = array())
     {
         $this->versions = $versions;
+        $this->parser = new Parser; 
```

Comment:
That's why I don't like PSR-2 - too much room for improvement. Anyway please make it consistent across the code. If in all other places the `()` are not added, then don't added it here too.